src/features/DrugChart/DrugChart.jsx

```diff
@@ -0,0 +1,231 @@
+import React from "react";
+import PropTypes from "prop-types";
+import moment from "moment";
+import Calendar from "../Calendar/Calendar";
+import CalendarHeader from "../CalendarHeader/CalendarHeader";
+import "./DrugChart.scss";
+
+export const TransformDrugChartData = (drugChartData) => {
+  const transformedDrugChartData = [];
+  drugChartData.forEach((schedule) => {
+    const { slots } = schedule;
+    const slotData = {};
+    slots.forEach((slot) => {
```

Comment:
Another coding nitpick, although this one is less of an issue. Here you are processing an array into an object. that is the use case for a .reduce() function. If you are comfortable with reduce, you can use that else let this one be.